shell/browser/electron_browser_context.cc

```diff
@@ -679,4 +718,21 @@ ElectronBrowserContext* ElectronBrowserContext::From(
   return new_context;
 }
 
+ElectronBrowserContext* ElectronBrowserContext::FromPath(
+    const base::FilePath path,
+    base::Value::Dict options) {
+  PartitionKey key(path.AsUTF8Unsafe(), false);
```

Comment:
This would make `session.fromPartition('/usr')` and  `session.fromPath('/usr')` return the same object which I think is a surprising behavior. You can probably add another property to `PartitionKey` to fix this.

Reply:
Hello @zcbenz ,

Thanks for reviewing the code. Currently session.fromPartition('/usr') would return an in memory session, and session.fromPath('/usr') would return a absolute path based session. 

However, you are absolutely right, in that if a session was created from using session.fromPartition("persist:test5") and session.fromPath("/home/user/.config/Electron/Partitions/test5) (where the sessions are being created by the same user and the default app Electron path is /home/user/.config/Electron), then the same browser context would be returned. I did this so as to avoid any risk of one sessions' data store tainting another sessions' data store, since I was not sure of the exact nature in which multiple browser windows manipulate the same partition.

I am happy to make the change if this should be the case. Do let me know.
I will add the other changes needed with this change.
Thanks,
Warm Regards,
George.

Reply:
Hmm I didn't think about the case when user would intentionally get the same session.

I am good with either behavior, but if we let `session.fromPartition` and `session.fromPath` return the same session, it should be documented and has some tests added. Just always returning different sessions feel lower maintenance.